lib/src/components/NxTransferList/__tests__/NxTransferList.test.tsx

```diff
@@ -23,348 +20,1072 @@ describe('NxTransferList', function() {
+              availableGroup = view.getByRole('group', { name: 'Available Items' }),
+              checkboxOne = view.getByRole('checkbox', { name: 'One' }),
+              checkboxTwo = view.getByRole('checkbox', { name: 'Two' }),
+              checkboxThree = view.getByRole('checkbox', { name: 'Three' });
 
-    expect(onChange).not.toHaveBeenCalled();
+          expect(availableGroup).toContainElement(checkboxOne);
+          expect(availableGroup).not.toContainElement(checkboxTwo);
+          expect(availableGroup).not.toContainElement(checkboxThree);
 
-    component.childAt(0).simulate('moveAll', [3]);
+          expect(checkboxOne).not.toBeChecked();
+        }
+    );
 
-    expect(onChange).toHaveBeenCalledWith(new Set([1, 3]));
+    it('renders the checkboxes for items present in selectedItems checked and in the "Transferred Items" group',
+        function() {
+          const view = quickRender({ allItems, selectedItems: new Set([2, 3]) }),
+              selectedGroup = view.getByRole('group', { name: 'Transferred Items' }),
+              checkboxOne = view.getByRole('checkbox', { name: 'One' }),
+              checkboxTwo = view.getByRole('checkbox', { name: 'Two' }),
+              checkboxThree = view.getByRole('checkbox', { name: 'Three' });
+
+          expect(selectedGroup).not.toContainElement(checkboxOne);
+          expect(selectedGroup).toContainElement(checkboxTwo);
+          expect(selectedGroup).toContainElement(checkboxThree);
+
+          expect(checkboxTwo).toBeChecked();
+          expect(checkboxThree).toBeChecked();
+        }
+    );
   });
 
-  it('calls onChange with the selected items set without removed ids when the second half\'s moveAll is clicked',
-      function() {
-        const onChange = jest.fn(),
-            allItems = [{ id: 1, displayName: 'foo' }, { id: 2, displayName: 'bar' }, { id: 3, displayName: 'qwerty' }],
-            selectedItems = new Set([1, 2]),
-            component = getShallow({ onChange, allItems, selectedItems, showMoveAll: true });
+  describe('checkbox interaction', function() {
+    it('fires onChange without the checkbox\'s corresponding id when a selected checkbox is clicked',
+        async function() {
+          const user = userEvent.setup(),
+              onChange = jest.fn(),
+              view = quickRender({ allItems, selectedItems: new Set([1, 3, 2]), showMoveAll: true, onChange }),
+              checkbox3 = view.getByRole('checkbox', { name: 'Three' });
 
-        expect(onChange).not.toHaveBeenCalled();
+          await user.click(checkbox3);
 
-        component.childAt(1).simulate('moveAll', [1]);
+          expect(onChange).toHaveBeenCalledTimes(1);
+          expect(onChange).toHaveBeenCalledWith(new Set([1, 2]));
+        }
+    );
+
+    it('fires onChange with the checkbox\'s corresponding id added to the end of the list when a selected ' +
+        'checkbox is clicked', async function() {
```

Comment:
Should it be 'fires onChange with the checkbox\'s corresponding id added to the end of the list when a checkbox is clicked'? Since the checkbox1 is not selected.

Reply:
> Should it be 'fires onChange with the checkbox's corresponding id added to the end of the list when a checkbox is clicked'? Since the checkbox1 is not selected.

Test name improved in 6585e49b3